test/react_test_utils_test.dart

```diff
@@ -229,6 +229,22 @@ void main() {
     expect(results[1]['tagName'], equals('DIV'));
   });
 
+  test('scryRenderedDomComponentsWithId', () {
+    var idName = 'searchableId';
+
+    component = renderIntoDocument(div({}, [
+      div({'id': idName}),
+      div({'id': idName}),
+      span({})
+    ]));
+
+    var results = scryRenderedDOMComponentsWithId(component, idName);
+
+    expect(results.length, 2);
+    expect(results[0]['props']['id'], equals(idName));
+    expect(results[1]['props']['id'], equals(idName));
+  });
```

Comment:
It would also be a good idea to test the scenario where the query returns an empty list.